### ISSUE TITLE
SharpLab IL Rewrite error

### ISSUE THREAD

**Issue by bernd5** [ashmind/SharpLab#1236]
Execute the following code:

```csharp
using System;

Algo.Foo<A>();
Algo.Foo<B>();
Algo.Foo<C>();

class A : IDoer
{
    public static void Do()
    {
        Console.WriteLine("A");
    }
}

class B : A, IDoer
{
    public new static void Do()
    {
        Console.WriteLine("B");
    }
}

class C : B, IDoer
{
    static void IDoer.Do()
    {
        Console.WriteLine("C");
    }
    public new static void Do()
    {
        Console.WriteLine("CDirect");
    }
}

interface IDoer
{
    public abstract static void Do();
}

class Algo
{
    public static void Foo<T>() where T : IDoer
    {
        T.Do();
    }
}
```

## Expected result:

A
B
C

## Actual result:

> Note: This exception is likely caused by SharpLab itself, and not the C# compiler.
> 
> Try adding [assembly: SharpLab.Runtime.NoILRewriting] to your code.
> If exception disappears, this is definitely a SharpLab issue, and should be [reported as such](https://github.com/ashmind/SharpLab/issues).
> System.BadImageFormatException: Bad IL format.
>    at Algo.Foo[T]()
>    at Program.<Main>$(String[] args)
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
>    at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)

**Comment by ashmind** [comment]
Thanks for reporting. Which branch did you see this on?
It should be fixed by #1230 in Default and newer branches, e.g. this works:
https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQ6CCAbA5gewDoAxffAHkwD4AKASgG4s8jSKAhWx5gks8gMJcmGNIgAEmcSHEBJACL4ApmHQBvdOK3i4AZh1IAbDoAs4xfU3aNabXYMBOGgCJMz7re0BfdD9ET2aUkAGjlFFXUrLT1xKCUAdwNjODMLOijxG3ttOCQnZ3Z3ETs/P3QJASD2UIVlVTQsnKNTMLrCNIzG7Nz8gSKMvzsYuMTc5NT8S08tLvselwF5AEswJQBjYH7p8VL0dCWoYBUAMwBDNaVWiIaMmNOAIwBnYDBz4CSWtJEysQRJFki2xiYxabHIABUuOJ4gALFSXcFBWrXOyzOzg9qTDwlXxAA==

**Comment by bernd5** [comment]
Not sure, but I think it was main.
Currently main it is not working (connection lost, reconnecting…)

**Comment by ashmind** [comment]
Yep main is broken for unrelated reasons (unfortunately I can only fix it in around a day, need to wait for a certain Roslyn build channel to build first).

**Comment by ashmind** [comment]
Confirmed as working in `main` now -- likely resolved by #1230.
I'll close the ticket, but please let me know if you see it again and I can reopen.
